src/statmodels.jl

```diff
@@ -213,6 +220,12 @@ Return the model response (a.k.a. the dependent variable).
 """
 model_response(obj::RegressionModel) = error("model_response is not defined for $(typeof(obj)).")
 
+"""
+    model_matrix(obj::RegressionModel)
+
+Return the model matrix (a.k.a. the design matrix).
+"""
+model_matrix(obj::RegressionModel) = error("model_matrix is not defined for $(typeof(obj)).")
```

Comment:
@andreasnoack suggested getting rid of the underscore (`modelmatrix`). I guess we can do that, and deprecate `model_response` later.

Reply:
Went ahead and changed it to `modelmatrix`. Is `model_response` being depreciated in favor of `modelresponse` for the next release? I think the decision was taken some time ago, but don't know the timeline.

Reply:
We're still discussing the best naming policy for these methods (#280), so I'm not sure it's worth deprecating the function right now, but indeed it's likely to happen at some point.

Reply:
Actually, I suggested renaming it to just `response` in that issue. OTOH, calling `modelmatrix` just `matrix` would be weird. `designmatrix` could be an interesting alternative.

Reply:
I come from an R background with `model.matrix` and `model.response` so I might be a bit biased on the issue. I know the Julian naming conventions recommend not using `_` if it can be understood without (e.g., `completecases`). I like the model prefix for consistency, but I think `response` and `designmatrix` are a good alternative. However, I am not complete sure the design matrix concept is shared among all fields that might use it... I know econometrics uses it as the standard and in machine learning it is used even if it is not at the same level (e.g., model matrix is also commonly used in ML as well, but design matrix is understood). If design matrix doesn't exclude any practitioners to the best of your knowledge I would be in favor of `designmatrix`.

Reply:
I like both `modelmatrix` and `designmatrix`. In my previous field (clinical biostats) I think "model matrix" was the dominant term, though "design matrix" was equally well understood; no one would be confused by that term. As another data point, one of my classes in grad school used the term "design matrix" exclusively.

Reply:
Please don't confound a model matrix with a design matrix.  They are different and there is a very good reason that the term "model matrix" is used in R.

If you know the sequence of (formula + data) -> ModelFrame -> ModelMatrix the "design" is reflected in the ModelFrame.  If you have a categorical factor, it is a single categorical variable at the level of the design.  At the level of the model matrix categorical factors are expanded to multiple columns of contrasts, interactions are expanded, polynomial terms are expanded, ...

This is why it is useful to retain not only the numerical values in the model matrix but also information on how you got them.  The ModelFrame with its terms information was based on the R implementation as a way to do this.  It may not be the best way to do this in a language as flexible as Julia but the thing to avoid is ending up with nothing but a matrix of numbers and no idea where these numbers came from.

If you consider designing experiments in the sense of choosing the levels for each of the primary covariates at each run you can see that a design matrix doesn't involve interactions (if you know what the levels of factors A and B are you know the level of the interaction A&B) or contrasts or polynomial terms.

Reply:
Thanks @dmbates. Let's go with `modelmatrix` then. It's probably easier to find/understand anyway. (Though [Wikipedia](https://en.wikipedia.org/wiki/Design_matrix) doesn't appear to make a distinction between "model matrix" and "design matrix".)

Reply:
Sounds good. `modelmatrix` is suppose to return the actual matrix used when fitting the model which includes the matrix that has already been through complete cases, contrasts, expansions, interactions, weighted, transformed (e.g., error component models), and made full rank when applicable. Should the updated documentation after merging the pull request explicitly indicate the above?

Reply:
@nalimilan I agree that model matrix and design matrix are often used interchangeably including in many textbooks where "X" is described as the design matrix.  Some of us who are old enough to have had courses on experimental design when it was taught separately from regression prefer to make the distinction.  For example, a constant column (a column of 1's) makes perfect sense as part of the model but not as part of the design.

Reply:
> Should the updated documentation after merging the pull request explicitly indicate the above?

That surely wouldn't hurt, but better do that directly in this PR.